stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/imagepreview/ImagePreviewActivity.kt

```diff
@@ -200,24 +534,83 @@ public class ImagePreviewActivity : AppCompatActivity() {
                     color = ChatTheme.colors.textHighEmphasis
                 )
 
-                // TODO we need to open a gallery of images here
-//                Icon(
-//                    modifier = Modifier
-//                        .align(Alignment.CenterEnd)
-//                        .padding(8.dp),
-//                    imageVector = Icons.Default.Apps,
-//                    contentDescription = stringResource(id = R.string.photos),
-//                    tint = ChatTheme.colors.textHighEmphasis
-//                )
+                Icon(
+                    modifier = Modifier
+                        .align(Alignment.CenterEnd)
+                        .padding(8.dp)
+                        .clickable(
+                            interactionSource = remember { MutableInteractionSource() },
+                            indication = rememberRipple(),
+                            onClick = { imagePreviewViewModel.toggleGallery(isShowingGallery = true) }
+                        ),
+                    imageVector = Icons.Default.Apps,
+                    contentDescription = stringResource(id = R.string.stream_compose_image_preview_photos),
+                    tint = ChatTheme.colors.textHighEmphasis
+                )
             }
         }
     }
 
-    private fun onShareImageClick(message: Message, currentPage: Int) {
+    /**
+     * Builds the image options based on the given [message]. These options let the user interact more
+     * with the images they're observing.
+     *
+     * @param message The message that holds all the images.
+     * @return [List] of options the user can choose from, in the form of [ImagePreviewOption].
+     */
+    @Composable
+    private fun defaultImageOptions(message: Message): List<ImagePreviewOption> {
+        val user by imagePreviewViewModel.user.collectAsState()
+        val options = mutableListOf(
+            ImagePreviewOption(
+                title = stringResource(id = R.string.stream_compose_image_preview_reply),
+                titleColor = ChatTheme.colors.textHighEmphasis,
+                icon = Icons.Default.Reply,
+                iconColor = ChatTheme.colors.textHighEmphasis,
+                action = Reply(message)
+            ),
+            ImagePreviewOption(
+                title = stringResource(id = R.string.stream_compose_image_preview_show_in_chat),
+                titleColor = ChatTheme.colors.textHighEmphasis,
+                icon = Icons.Default.PanoramaFishEye,
```

Comment:
This icon looks quite weird to me, we should replace it with the one in our design